mlflow/tracking/fluent.py

```diff
@@ -80,20 +89,48 @@ def set_experiment(experiment_name: str) -> None:
         Tags: {}
         Lifecycle_stage: active
     """
-    client = MlflowClient()
-    experiment = client.get_experiment_by_name(experiment_name)
-    exp_id = experiment.experiment_id if experiment else None
-    if exp_id is None:  # id can be 0
-        print("INFO: '{}' does not exist. Creating a new experiment".format(experiment_name))
-        exp_id = client.create_experiment(experiment_name)
-    elif experiment.lifecycle_stage == LifecycleStage.DELETED:
+    if (experiment_name is not None and experiment_id is not None) or (
+        experiment_name is None and experiment_id is None
+    ):
         raise MlflowException(
-            "Cannot set a deleted experiment '%s' as the active experiment."
-            " You can restore the experiment, or permanently delete the "
-            " experiment to create a new one." % experiment.name
+            message="Must specify exactly one of: `experiment_id` or `experiment_name`.",
+            error_code=INVALID_PARAMETER_VALUE,
         )
+
+    def verify_experiment_active(experiment):
+        if experiment.lifecycle_stage == LifecycleStage.DELETED:
+            raise MlflowException(
+                message=(
+                    "Cannot set a deleted experiment '%s' as the active experiment."
+                    " You can restore the experiment, or permanently delete the "
+                    " experiment to create a new one." % experiment.name
+                ),
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+    client = MlflowClient()
+    if experiment_id is None:
+        experiment = client.get_experiment_by_name(experiment_name)
+        if experiment:
+            verify_experiment_active(experiment)
```

Comment:
How about verify outside of the if-else block?

Reply:
I think we might need to check if experiment is present to check outside of the if block or we need to replicate that if experiment check in the `verify_experiment_active` function as far as I can tell.

Reply:
The challenge here is that we'd need to re-fetch the experiment in order to perform the activity check in the case where a new experiment was created, since `client.create_experiment` returns an experiment ID, rather than the whole experiment entity. In the interest of saving a network call, I've embedded `verify_experiment_active` within a couple interior branches.